Realm.Shared/handles/LinkListHandle.cs

```diff
@@ -15,19 +15,100 @@
 // limitations under the License.
 //
 ////////////////////////////////////////////////////////////////////////////
- 
+
+using System;
+using System.Runtime.InteropServices;
+
 namespace Realms
 {
     internal class LinkListHandle:RealmHandle
     {
+        private static class NativeMethods
+        {
+            [DllImport(InteropConfig.DLL_NAME, EntryPoint = "linklist_add", CallingConvention = CallingConvention.Cdecl)]
+            public static extern void add(LinkListHandle linklistHandle, IntPtr row_ndx, out NativeException ex);
+
+            [DllImport(InteropConfig.DLL_NAME, EntryPoint = "linklist_insert", CallingConvention = CallingConvention.Cdecl)]
+            public static extern void insert(LinkListHandle linklistHandle, IntPtr link_ndx, IntPtr row_ndx, out NativeException ex);
+
+            [DllImport(InteropConfig.DLL_NAME, EntryPoint = "linklist_erase", CallingConvention = CallingConvention.Cdecl)]
+            public static extern void erase(LinkListHandle linklistHandle, IntPtr row_ndx, out NativeException ex);
+
+            [DllImport(InteropConfig.DLL_NAME, EntryPoint = "linklist_clear", CallingConvention = CallingConvention.Cdecl)]
+            public static extern void clear(LinkListHandle linklistHandle, out NativeException ex);
+
+            [DllImport(InteropConfig.DLL_NAME, EntryPoint = "linklist_get", CallingConvention = CallingConvention.Cdecl)]
+            public static extern IntPtr get(LinkListHandle linklistHandle, IntPtr link_ndx, out NativeException ex);
+
+            [DllImport(InteropConfig.DLL_NAME, EntryPoint = "linklist_find", CallingConvention = CallingConvention.Cdecl)]
+            public static extern IntPtr find(LinkListHandle linklistHandle, IntPtr link_ndx, IntPtr start_from, out NativeException ex);
+
+            [DllImport(InteropConfig.DLL_NAME, EntryPoint = "linklist_size", CallingConvention = CallingConvention.Cdecl)]
+            public static extern IntPtr size(LinkListHandle linklistHandle, out NativeException ex);
+
+            [DllImport(InteropConfig.DLL_NAME, EntryPoint = "linklist_destroy", CallingConvention = CallingConvention.Cdecl)]
+            public static extern void destroy(IntPtr linklistInternalHandle);
+        }
+
         internal LinkListHandle(RealmHandle root) : base(root)
         {
         }
 
         protected override void Unbind()
         {
-            NativeLinkList.destroy(handle);
+            NativeMethods.destroy(handle);
```

Comment:
Can we get an exception here?
I am a bit suspicious of any native calls now which do not have the exception handling, as implicitly they all had it by the previous callback mechanism. It seems to me that because we're destroying reference counted pointers, this could trigger quite a lot of work in core in some cases